package/MDAnalysis/analysis/diffusionmap.py

```diff
@@ -243,12 +244,20 @@ class DistanceMatrix(AnalysisBase):
     .. versionchanged:: 2.0.0
          :attr:`dist_matrix` is now stored in a
          :class:`MDAnalysis.analysis.base.Results` instance.
-
+    .. versionchanged:: 2.2.0
+         :class:`DistanceMatrix` now also accepts `AtomGroup`.
     """
     def __init__(self, universe, select='all', metric=rmsd, cutoff=1E0-5,
                  weights=None, **kwargs):
         # remember that this must be called before referencing self.n_frames
-        super(DistanceMatrix, self).__init__(universe.trajectory, **kwargs)
+        super(DistanceMatrix, self).__init__(universe.universe.trajectory,
+                                             **kwargs)
+
+        if isinstance(universe, UpdatingAtomGroup):
+            wmsg = ("U must be a static AtomGroup. Parsing an updating AtomGroup "
+                    "will result in a static AtomGroup with the first frame "
```

Comment:
Please double check this, but I'm pretty sure it'll freeze it at the current frame, so if someone did:

```python
u.trajectory[10]
uag = u.select_atoms('resname HOH and within 4 resname LIG', updating=True)
dmatrix = DistanceMatrix(ag)
```

Then you're doing uag.select_atoms('all') on the current frame, rather than the first frame.

```suggestion
                    "will result in a static AtomGroup with the current frame "
```

Reply:
yes. you're right. It's the current selection.